src/App.js

```diff
@@ -3,22 +3,28 @@ import { useEffect } from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = useState(() => 
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  );
+function App() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList))
+    new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({ data: { todoList: JSON.parse(localStorage.getItem('savedTodoList')) || [] } }); 
+      }, 2000); 
+    })
+    .then((result) => {
+      setTodoList(result.data.todoList);
+      setIsLoading(false);
+    });
+  }, []); 
 
+  useEffect(() => {
+    if (isLoading === false) {
+      localStorage.setItem('savedTodoList', JSON.stringify(todoList));
+    }
```

Comment:
You usually want to make use of thruthy or falsy values as if conditions.
Example:
```
let string = "";
if(string){
  statement1
} else {
  statement2
}
```

Since string variable is falsy (falsy is a value that is considered false when encountered in a boolean context), statement2 is executed
Falsy values:

- null
- undefined
- false
- 0
- ""
- NaN

In contrast a truthy value is a value that is considered true when encountered in a boolean context. All values are truthy unless they are defined as falsy (list above).

So in line 23 is suffice to say that:
```
if(!isLoading) {
      localStorage.setItem('savedTodoList', JSON.stringify(todoList));
}
```